applications/nrf_desktop/src/modules/dfu_mcumgr.c

```diff
@@ -75,6 +75,43 @@ static void dfu_timeout_handler(struct k_work *work)
 	}
 }
 
+static bool smp_cmd_is_suit_dfu_cmd(const struct mgmt_evt_op_cmd_arg *cmd,
+				    const char **smp_cmd_name)
+{
+#if CONFIG_MGMT_SUITFU_GRP_SUIT
+	if (cmd->group == CONFIG_MGMT_GROUP_ID_SUIT) {
+		*smp_cmd_name = "SUIT Management";
+		return true;
+	}
+#endif /* CONFIG_MGMT_SUITFU_GRP_SUIT */
+
+	return false;
+}
+
+static bool smp_cmd_is_dfu_cmd(const struct mgmt_evt_op_cmd_arg *cmd)
+{
+	const char *smp_cmd_name = "Unknown";
+	bool dfu_transfer_cmd = true;
+
+	if (cmd->group == MGMT_GROUP_ID_IMAGE) {
+		smp_cmd_name = "Image Management";
+	} else if ((cmd->group == MGMT_GROUP_ID_OS) && (cmd->id == OS_MGMT_ID_RESET)) {
+		smp_cmd_name = "OS Management Reset";
+	} else {
+		dfu_transfer_cmd = false;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_BACKEND_SUIT) && !dfu_transfer_cmd) {
+		dfu_transfer_cmd = smp_cmd_is_suit_dfu_cmd(cmd, &smp_cmd_name);
+	}
```

Comment:
nit that can be ignored: instead of checking `!dfu_transfer_cmd` here, couldn't it be as last `else if` in the chain above? :D

Reply:
I am reluctant to do it, as this statement is not mutually exclusive with conditions in the `if else` chain. If had it in the `if else` chain` and you added another condition that checks specific group and command ID as the next check, you would get faulty results for the SUIT backend Kconfig enabled. This new condition would never have been entered as `IS_ENABLED(CONFIG_DESKTOP_DFU_BACKEND_SUIT)` would always return true.